### PR TITLE
Bump FreeRadius 3.0.18 and bugfix

### PR DESCRIPTION
**Pull request by geaaru**
* Drop v.3.0.15
* Fix issue #675230
* Bump v.3.0.18

### CODE REVIEW

File: net-dialup/freeradius/freeradius-3.0.18.ebuild [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+		install
+
+	fowners -R root:radius /etc/raddb
+	fowners -R radius:radius /var/log/radius
+
+	pamd_mimic_system radiusd auth account password session
+
+	dodoc CREDITS
+
+	rm "${D}/usr/sbin/rc.radiusd" || die
```
Comment (monsieurp) [inline net-dialup/freeradius/freeradius-3.0.18.ebuild]: can you perform this operation at the very beginning of the `src_install` function?

File: net-dialup/freeradius/freeradius-3.0.18.ebuild [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+	newinitd "${FILESDIR}/radius.init-r3" radiusd
+	newconfd "${FILESDIR}/radius.conf-r4" radiusd
+
+	systemd_newtmpfilesd "${FILESDIR}"/freeradius.tmpfiles freeradius.conf
+	systemd_dounit "${FILESDIR}"/freeradius.service
+
+	find "${ED}" \( -name "*.a" -o -name "*.la" \) -delete || die
```
Comment (monsieurp) [inline net-dialup/freeradius/freeradius-3.0.18.ebuild]: Same remark.

File: net-dialup/freeradius/freeradius-3.0.18.ebuild [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+}
+
+pkg_config() {
+	if use ssl; then
+		cd "${ROOT}"/etc/raddb/certs
+		./bootstrap
+
+		chown -R root:radius "${ROOT}"/etc/raddb/certs
```
Comment (monsieurp) [inline net-dialup/freeradius/freeradius-3.0.18.ebuild]: why not using `fowners` here?

File: net-dialup/freeradius/freeradius-3.0.18.ebuild (hunk 1 above)
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.18.ebuild]: ok, np.

File: net-dialup/freeradius/freeradius-3.0.18.ebuild (hunk 2 above)
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.18.ebuild]: ok

File: net-dialup/freeradius/freeradius-3.0.18.ebuild (hunk 3 above)
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.18.ebuild]: :) probably it's a legacy row of the previous maintainer. I will fix it. Thanks

File: net-dialup/freeradius/freeradius-3.0.18.ebuild (hunk 2 above)
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.18.ebuild]: what do you think if I replace `find "${ED}"...` with `prune_libtool_files`?

Comment (monsieurp) [reply net-dialup/freeradius/freeradius-3.0.18.ebuild]: :+1:

File: net-dialup/freeradius/freeradius-3.0.18.ebuild [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+	newinitd "${FILESDIR}/radius.init-r3" radiusd
+	newconfd "${FILESDIR}/radius.conf-r4" radiusd
+
+	prune_libtool_files
+}
+
+pkg_config() {
+	if use ssl; then
+		cd "${ROOT}"/etc/raddb/certs
+		./bootstrap
```
Comment (monsieurp) [inline net-dialup/freeradius/freeradius-3.0.18.ebuild]: Is this a shell script? I believe so. In this case we need to catch errors if something goes wrong. Please add a `|| die` here as well.

File: net-dialup/freeradius/freeradius-3.0.18.ebuild [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+	newinitd "${FILESDIR}/radius.init-r3" radiusd
+	newconfd "${FILESDIR}/radius.conf-r4" radiusd
+
+	prune_libtool_files
+}
+
+pkg_config() {
+	if use ssl; then
+		cd "${ROOT}"/etc/raddb/certs
```
Comment (monsieurp) [inline net-dialup/freeradius/freeradius-3.0.18.ebuild]: Missing `|| die` here`.

File: net-dialup/freeradius/freeradius-3.0.18.ebuild (hunk 4 above)
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.18.ebuild]: Yes, it is a sh script that generates certificates. Ok for `|| die`

File: net-dialup/freeradius/freeradius-3.0.18.ebuild (hunk 5 above)
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.18.ebuild]: done

**Comment by stikonas** [comment]
Why was stable version dropped? I see no explanation in commit message.

**Comment by geaaru** [comment]
Because there are a lot of bugfixes in the releases 3.0.{16,17,18} and I don't consider 3.0.15 as stable. I move 3.0.17 (that I have in production from at least one year) as stable.
I try to create stabilization issue in these days.

**Pull request closed without merging**